File: ABC191/191D.py
import math

X, Y, R = map(float, input().split())

a = math.ceil(X - R)
b = math.floor(X + R)
k = 0

for x in range(a, b+1):
    z = math.sqrt((R + x - X) * (R - x + X))
    y_up = math.floor(Y + z)
    y_down = math.ceil(Y - z)
    k += y_up - y_down + 1

print(k)

File: ABC/ABC191/191D.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    double X, Y, R;
    cin >> X >> Y >> R;
    int a, b, k;
    a = ceil(X -R);
    b = floor(X + R);
    k = 0;
    for (int x = a; x < b+1; x++) {
        int y_up, y_down;
        y_up = floor(Y + sqrt((R + x - X) * (R - x + X)));
        y_down = ceil(Y - sqrt((R + x - X) * (R - x + X)));
        k += y_up - y_down + 1;
    }
    cout << k << endl;
}